pkg/server/endpoints/authorized_entryfetcher_registration_entries.go

```diff
@@ -218,32 +222,44 @@ func (a *registrationEntries) updateCache(ctx context.Context) error {
 
 // updateCacheEntry update/deletes/creates an individual registration entry in the cache.
 func (a *registrationEntries) updateCachedEntries(ctx context.Context) error {
-	for entryId := range a.fetchEntries {
-		commonEntry, err := a.ds.FetchRegistrationEntry(ctx, entryId)
+	entryIds := slices.Collect(maps.Keys(a.fetchEntries))
+	for pageStart := 0; pageStart < len(entryIds); pageStart += int(a.pageSize) {
+		fetchEntries := a.fetchEntriesPage(entryIds, pageStart)
+		commonEntries, err := a.ds.FetchRegistrationEntries(ctx, fetchEntries)
 		if err != nil {
-			continue
+			return err
 		}
 
-		if commonEntry == nil {
-			a.cache.RemoveEntry(entryId)
-			delete(a.fetchEntries, entryId)
-			continue
-		}
+		for _, entryId := range fetchEntries {
+			commonEntry := commonEntries[entryId]
+			if commonEntry == nil {
```

Comment:
nit: could we check for the presence of entry, I think we won't have `nil` entries in the map, they would just not be in the map:
```suggestion
			commonEntry, ok := commonEntries[entryId]
			if !ok {
```